homeassistant/components/zha/core/channels/hvac.py

```diff
@@ -35,41 +67,390 @@ async def async_set_speed(self, value) -> None:
+    def occupancy(self) -> Optional[int]:
+        """Is occupancy detected."""
+        return self._occupancy
+
+    @property
+    def occupied_cooling_setpoint(self) -> Optional[int]:
+        """Temperature when room is occupied."""
+        return self._occupied_cooling_setpoint
+
+    @property
+    def occupied_heating_setpoint(self) -> Optional[int]:
+        """Temperature when room is occupied."""
+        return self._occupied_heating_setpoint
+
+    @property
+    def pi_cooling_demand(self) -> int:
+        """Cooling demand."""
+        return self._pi_cooling_demand
+
+    @property
+    def pi_heating_demand(self) -> int:
+        """Heating demand."""
+        return self._pi_heating_demand
+
+    @property
+    def running_mode(self) -> Optional[int]:
+        """Thermostat running mode."""
+        return self._running_mode
+
+    @property
+    def running_state(self) -> Optional[int]:
+        """Thermostat running state, state of heat, cool, fan relays."""
+        return self._running_state
+
+    @property
+    def system_mode(self) -> Optional[int]:
+        """System mode."""
+        return self._system_mode
+
+    @property
+    def unoccupied_cooling_setpoint(self) -> Optional[int]:
+        """Temperature when room is not occupied."""
+        return self._unoccupied_cooling_setpoint
+
+    @property
+    def unoccupied_heating_setpoint(self) -> Optional[int]:
+        """Temperature when room is not occupied."""
+        return self._unoccupied_heating_setpoint
+
+    @callback
+    def attribute_updated(self, attrid, value):
+        """Handle attribute update cluster."""
+        attr_name = self.cluster.attributes.get(attrid, [attrid])[0]
+        self.debug(
+            "Attribute report '%s'[%s] = %s", self.cluster.name, attr_name, value
+        )
+        setattr(self, f"_{attr_name}", value)
+        self.async_send_signal(
+            f"{self.unique_id}_{SIGNAL_ATTR_UPDATED}",
+            AttributeUpdateRecord(attrid, attr_name, value),
+        )
+
+    async def _chunk_attr_read(self, attrs, cached=False):
+        chunk, attrs = attrs[:4], attrs[4:]
+        while chunk:
+            res, fail = await self.cluster.read_attributes(chunk, allow_cache=cached)
+            self.debug("read attributes: Success: %s. Failed: %s", res, fail)
+            for attr in chunk:
+                self._init_attrs.pop(attr, None)
+                if attr in fail:
+                    continue
+                if isinstance(attr, str):
+                    setattr(self, f"_{attr}", res[attr])
+                self.async_send_signal(
+                    f"{self.unique_id}_{SIGNAL_ATTR_UPDATED}",
+                    AttributeUpdateRecord(None, attr, res[attr]),
+                )
+
+            chunk, attrs = attrs[:4], attrs[4:]
+
+    async def configure_reporting(self):
+        """Configure attribute reporting for a cluster.
+
+        This also swallows DeliveryError exceptions that are thrown when
```

Comment:
Not for this PR but I want to discuss this a bit. Starting to think we need to do some work on device joining / configuration based on recent issues.

Reply:
same as above. I wanna get this a more wider testing and gather feedback. 
Climate does configure reporting for multiple attributes, however some vendors still do not stick  to the spec in the response, so need to add some extra logic in response parsing.

But ATM, the only other cluster we use multiple attributes reporting is the light cluster. So for any other cluster this could be a drop-in replacement.